### PR TITLE
Remodel ScanSelection and MS level filter

### PR DESCRIPTION
**Pull request by robinschmid**
This is a quite big pull request to harmonize SubModuleParameter, ParameterSetParameter, OptionalModuleParameter

There are options now to either show a sub ParameterSet in a sub panel or in a dialog.

There is no way to auto size a Stage (dialog) so we estimate some size changes here based on the number of new parameters that are shown.

image

image

**Correlation grouping**
image

image

### CODE REVIEW

**Comment by robinschmid** [comment]
- Added support to load the scan selections from old parameter files
- Added string to describe the selected filters. Otherwise, people will miss that sometimes. Its only shown when selected.

image
image
image
image

File: src/main/java/io/github/mzmine/parameters/parametertypes/selectors/ScanSelectionComponent.java [hunk 1]
```diff
@@ -25,246 +25,72 @@
...
-      scanMobilityRange = null;
-      polarity = null;
-      spectrumType = null;
-      msLevel = null;
-      scanDefinition = null;
-      updateRestrictionList();
+import javafx.scene.text.Text;
 
-    });
+public class ScanSelectionComponent extends OptionalModuleComponent {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/parameters/parametertypes/selectors/ScanSelectionComponent.java:33]: OptionalModuleComponent -> Does this mean the scan selection always has a checkbox now?

File: src/main/java/io/github/mzmine/parameters/AbstractParameter.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.mzmine.parameters;
+
+import javafx.scene.Node;
+
+public abstract class AbstractParameter<ValueType, EditorComponent extends Node> implements
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/parameters/AbstractParameter.java:30]: Why do we need this now?

**Review by SteffenHeu — APPROVED** [review]
only a few comments

File: src/main/java/io/github/mzmine/parameters/parametertypes/selectors/ScanSelectionComponent.java (hunk 1 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/parameters/parametertypes/selectors/ScanSelectionComponent.java:33]: yeah its always with checkbox... could be changed but in general you always get the ScanSelection.ALL_SCANS if its deselected

**Pull request merged**